File: Problemas/Prime Time/p10200.py
def is_prime(num):
    if num % 2 == 0 or num % 3 == 0 or num % 5 == 0 or num % 7 == 0:
        return False
    
    for i in range(11, int(num**0.5)+1, 2):
        if num % i == 0:
            return False
        
    return True
 
def extendPrime(primes, num):
    aux = primes
    for i in range(len(primes), num+1):
        aux.append(is_prime(i))
    return aux

def fun(x):
    return x*x + x + 41
    
def main():
    primes = []
    
    try:
        while True:
            a, b = input().split(" ")
            a = int(a)
            b = int(b)
            maxV = fun(b)
            if  maxV > len(primes):
                print("Extendmos", len(primes), maxV)
                primes = extendPrime(primes, maxV)
                
            prime = 0
            for i in range(a, b+1):
                if primes[fun(i)]:
                    prime += 1
            print(((prime/ (b-a+1))) * 100)
            
    except EOFError:
        pass
        
if __name__ == "__main__":
    main()
    

File: Problemas/Prime Time/p10200.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <fstream>

using namespace std;

bool is_prime(int num) {
    if (num < 2) return false;
    if (num % 2 == 0 || num % 3 == 0 || num % 5 == 0 || num % 7 == 0) 
        return (num == 2 || num == 3 || num == 5 || num == 7);
    
    for (int i = 11; i * i <= num; i += 2) {
        if (num % i == 0)
            return false;
    }
    return true;
}

void extendPrime(vector<bool>& primes, int num) {
    int currentSize = primes.size();
    primes.resize(num + 1, false);
    
    for (int i = currentSize; i <= num; i++) {
        primes[i] = is_prime(i);
    }
}

int fun(int x) {
    return x * x + x + 41;
}

int main(int argc, char* argv[]) {

    string inputFileName = argv[1];
    string outputFileName = argv[2];

    ifstream inputFile(inputFileName);
    ofstream outputFile(outputFileName);

    vector<bool> primes;
    int a, b;

    while (inputFile >> a >> b) {
        int maxV = fun(b);
        
        if (maxV >= primes.size()) {
            extendPrime(primes, maxV);
        }

        int primeCount = 0;
        for (int i = a; i <= b; i++) {
            if (primes[fun(i)]) {
                primeCount++;
            }
        }

        double percentage = (double)primeCount / (b - a + 1) * 100;
        outputFile << fixed;
        outputFile.precision(2);
        outputFile << percentage << endl;
    }

    inputFile.close();
    outputFile.close();
    return 0;
}
